### DISCUSSION TITLE
beam search in rnnt?

### DISCUSSION THREAD

**Discussion by BuaaAlban** [espnet/espnet#4510] [Q&A] [answered]
https://github.com/espnet/espnet/blob/f2778f798b76e102602078213a21b63aa592be70/espnet/nets/beam_search_transducer.py#L284  Why do we ignore the first element when select topk ? m

https://github.com/espnet/espnet/blob/f2778f798b76e102602078213a21b63aa592be70/espnet/nets/beam_search_transducer.py#L288
Why do we add the first element of logp?
I think it corresponds to the Pr(y*) = Pr(y*)Pr(blank| y,t) of the algo in paper, shouldn't it be the logp of blank index?

**Comment by b-flo** [comment, marked as answer]
Hi,

> shouldn't it be the logp of blank index?

`logp[0]` is the log-probability of `blank` (ID 0 is reserved for blank in ESPnet). `topk` contains top K non-blank log probs.

**Reply by BuaaAlban** [reply]
Thank you for the explanation! 
But it might be better to avoid the hard code as there is an attribute "self.blank_id"

**Reply by b-flo** [reply]
You're right! This file was not updated for a while as we've moved to ESPnet2. The most updated version of the beam search interface for Transducer is in #4479 currently (which dropped the `blank_id` attribute).
I'll patch the ESPnet1 code when I have time, as there are other issues in this version I didn't fix.
